### ISSUE TITLE
EB Environment Variables from Github Secrets

### ISSUE THREAD

**Issue by chronos-ohm** [einaregilsson/beanstalk-deploy#81]
Any way to pass environment variables to EB through just setting up the secrets on Github and defining them in the workflow yml, just like the AWS secrets are set?

Would be better if all the secrets were contained in one place.

**Comment by moon-sam** [comment]
I agree that they would be nice to have in one place, but one argument against that is there's no reason for github to ever know these secrets, since they aren't needed for deploy. They can be configured permanently in the AWS Beanstalk environment.

**Comment by jleguina** [comment]
Any updates on this? Cheers

**Comment by gbhand** [comment]
Sadly Einar is no longer with us https://github.com/einaregilsson/beanstalk-deploy/issues/93#issuecomment-1315173464 but @gzur has taken over maintaining the repo and is still actively reviewing PRs as they come in.

Environment variables could be set in this action in the [`UpdateEnvironment` API call](https://docs.aws.amazon.com/elasticbeanstalk/latest/api/API_UpdateEnvironment.html) by adding environment variables to the `OptionSettings` request parameter.

Environment variables can also be set by including an [.ebextensions config file](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/ebextensions-optionsettings.html) with placeholder values filled in by GitHub Actions. Additionally for applications on the Docker platform you can override all environment variables by [including a `.env` file in your application package](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/create_deploy_docker.container.console.html).

**Comment by ameya-shahu** [comment]
Hi @gbhand , Could you please provide me some sample script?

**Comment by vitalykarasik** [comment]
I'm generating my '.env' from GitHub secret as simple as 
```
echo '${{ secrets.ENV_FILE }}' > .env
```
